lib/grapevine_web/live/post_live.ex

```diff
@@ -29,6 +31,7 @@ defmodule GrapevineWeb.PostLive do
   def mount(_params, _session, socket) do
     posts = Posts.show_all()
     categories = Posts.get_categories()
+    categories = [%{id: 0, name: "All"}] ++ categories
```

Comment:
```suggestion
    categories = [%{id: 0, name: "All"} | categories]
```

https://elixirschool.com/en/lessons/basics/collections/#lists-0

Reply:
Sometimes we have to concatenate a list by using `++`. Whenever possible, using `[element | list ]` will be even better in terms of speed.

Reply:
The way we are using it right now it is a prepend element.

If you think about it the smallest one comes first.

 The memory consumed will be the same.

The tail for this list here: `[%{id: 0, name: "All"}]` it point it out to an empty list, so, when I concat with the biggest ones, I will point to a new list instead of an empty list, and we will not put the entire list in memory to concat them.